<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>vntEvaluacion</class>
 <widget class="QMainWindow" name="vntEvaluacion">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>579</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Sistema de Gestión de Evaluaciones</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_main">
    <item>
     <widget class="QLabel" name="lblTitulo">
      <property name="font">
       <font>
        <pointsize>16</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Sistema de Gestión de Evaluaciones Académicas</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_crear">
       <attribute name="title">
        <string>Crear Evaluación</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_crear">
        <item>
         <widget class="QGroupBox" name="groupBox_datos">
          <property name="title">
           <string>Datos de la Evaluación</string>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="lblNombre">
             <property name="text">
              <string>Nombre:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="txtNombre">
             <property name="placeholderText">
              <string>Ingrese el nombre de la evaluación</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="lblFecha">
             <property name="text">
              <string>Fecha:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDateEdit" name="dateEdit_fecha">
             <property name="calendarPopup">
              <bool>true</bool>
             </property>
             <property name="date">
              <date>
               <year>2024</year>
               <month>1</month>
               <day>1</day>
              </date>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="lblPuntaje">
             <property name="text">
              <string>Puntaje:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_puntaje">
             <property name="maximum">
              <double>100.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.500000000000000</double>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="lblTipoEvaluacion">
             <property name="text">
              <string>Tipo de Evaluación:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QComboBox" name="cbTipo">
             <item>
              <property name="text">
               <string>Examen</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Trabajo</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Presentación</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_parametros">
          <property name="title">
           <string>Parámetros Específicos</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_parametros">
           <item>
            <widget class="QStackedWidget" name="stackedWidget_parametros">
             <property name="currentIndex">
              <number>1</number>
             </property>
             <widget class="QWidget" name="page_examen">
              <layout class="QFormLayout" name="formLayout_examen">
               <item row="0" column="0">
                <widget class="QLabel" name="lblDuracion">
                 <property name="text">
                  <string>Duración (minutos):</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSpinBox" name="spinBox_duracion">
                 <property name="minimum">
                  <number>15</number>
                 </property>
                 <property name="maximum">
                  <number>300</number>
                 </property>
                 <property name="value">
                  <number>120</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="lblPreguntas">
                 <property name="text">
                  <string>Número de preguntas:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSpinBox" name="spinBox_preguntas">
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>100</number>
                 </property>
                 <property name="value">
                  <number>20</number>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="page_trabajo">
              <layout class="QFormLayout" name="formLayout_trabajo">
               <item row="0" column="0">
                <widget class="QLabel" name="lblPaginas">
                 <property name="text">
                  <string>Número de páginas:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSpinBox" name="spinBox_paginas">
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>100</number>
                 </property>
                 <property name="value">
                  <number>10</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_tema">
                 <property name="text">
                  <string>Tema:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="txtTema">
                 <property name="placeholderText">
                  <string>Ingrese el tema del trabajo</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="page_presentacion">
              <layout class="QFormLayout" name="formLayout_presentacion">
               <item row="0" column="0">
                <widget class="QLabel" name="lblDuracionPres">
                 <property name="text">
                  <string>Duración (minutos):</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSpinBox" name="spinBox_duracion_pres">
                 <property name="minimum">
                  <number>5</number>
                 </property>
                 <property name="maximum">
                  <number>60</number>
                 </property>
                 <property name="value">
                  <number>15</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="lblAudiencia">
                 <property name="text">
                  <string>Tamaño de audiencia:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSpinBox" name="spinBox_audiencia">
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>100</number>
                 </property>
                 <property name="value">
                  <number>25</number>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_botones_crear">
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="btnLimpiar">
            <property name="text">
             <string>Limpiar</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnCrear">
            <property name="text">
             <string>Crear Evaluación</string>
            </property>
            <property name="default">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_lista">
       <attribute name="title">
        <string>Lista de Evaluaciones</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_lista">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_filtros">
          <item>
           <widget class="QLabel" name="lblFiltro">
            <property name="text">
             <string>Filtrar por tipo:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cbFiltro">
            <item>
             <property name="text">
              <string>Todos</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Examen</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Trabajo</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Presentación</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="btnActualizar">
            <property name="text">
             <string>Actualizar</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableWidget" name="tableWidget_evaluaciones">
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
          </property>
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <column>
           <property name="text">
            <string>Nombre</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Tipo</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Fecha</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Puntaje</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Nota Calculada</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_botones_lista">
          <item>
           <widget class="QPushButton" name="btnVerDetalle">
            <property name="text">
             <string>Ver Detalle</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnEditar">
            <property name="text">
             <string>Editar</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnEliminar">
            <property name="text">
             <string>Eliminar</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_estadisticas">
       <attribute name="title">
        <string>Estadísticas</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_estadisticas">
        <item>
         <widget class="QGroupBox" name="groupBox_resumen">
          <property name="title">
           <string>Resumen General</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_resumen">
           <item row="0" column="0">
            <widget class="QLabel" name="lblTotalEval">
             <property name="text">
              <string>Total de Evaluaciones:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="lblTotalNumero">
             <property name="font">
              <font>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="lblPromedio">
             <property name="text">
              <string>Promedio General:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="lblPromedioNumero">
             <property name="font">
              <font>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>0.0</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="lblMejor">
             <property name="text">
              <string>Mejor Evaluación:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="lblMejorNombre">
             <property name="font">
              <font>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>N/A</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_por_tipo">
          <property name="title">
           <string>Estadísticas por Tipo</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_tipos">
           <item>
            <widget class="QTextEdit" name="textEdit_estadisticas">
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="html">
              <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'MS Shell Dlg 2'; font-size:8pt;&quot;&gt;No hay evaluaciones registradas aún.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>33</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuArchivo">
    <property name="title">
     <string>Archivo</string>
    </property>
    <addaction name="actionNuevo"/>
    <addaction name="actionAbrir"/>
    <addaction name="actionGuardar"/>
    <addaction name="separator"/>
    <addaction name="actionSalir"/>
   </widget>
   <widget class="QMenu" name="menuAyuda">
    <property name="title">
     <string>Ayuda</string>
    </property>
    <addaction name="actionAcerca_de"/>
   </widget>
   <addaction name="menuArchivo"/>
   <addaction name="menuAyuda"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNuevo">
   <property name="text">
    <string>Nuevo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionAbrir">
   <property name="text">
    <string>Abrir</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionGuardar">
   <property name="text">
    <string>Guardar</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSalir">
   <property name="text">
    <string>Salir</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAcerca_de">
   <property name="text">
    <string>Acerca de</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>cbTipo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackedWidget_parametros</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
